src/main/scala/is/hail/variant/MatrixTable.scala

```diff
@@ -1151,7 +1151,12 @@ class MatrixTable(val hc: HailContext, val ast: MatrixIR) {
           }
         case Apply(_, "annotate", Array(SymRef(_, "va"), newstruct)) =>
           coerce[TStruct](newstruct.`type`).fieldNames.toSet.intersect(rowKey.toSet).nonEmpty
-        case _ => true
+        case x@Apply(_, "drop", Array(SymRef(_, "va"), _*)) =>
```

Comment:
can this be `x@Apply(_, "drop", SymRef(_, "va") +: droppedFields)` for more clarity? I saw that syntax used in a pattern match once but not super sure how it works

Reply:
oops, saw this after you approved

Reply:
I didn't even know that was a valid match syntax!